community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -285,41 +287,20 @@ boolean applicable( DiagnosticsPhase phase )
      * refactored into bigger components that wrap the very granular things we depend on here.
      */
-    public NeoStoreDataSource(
-            File storeDir,
-            Config config,
-            IdGeneratorFactory idGeneratorFactory,
-            LogService logService,
-            JobScheduler scheduler,
-            TokenNameLookup tokenNameLookup,
-            DependencyResolver dependencyResolver,
-            PropertyKeyTokenHolder propertyKeyTokens,
-            LabelTokenHolder labelTokens,
-            RelationshipTypeTokenHolder relationshipTypeTokens,
-            StatementLocksFactory statementLocksFactory,
-            SchemaWriteGuard schemaWriteGuard,
-            TransactionEventHandlers transactionEventHandlers,
-            IndexingService.Monitor indexingServiceMonitor,
-            FileSystemAbstraction fs,
-            TransactionMonitor transactionMonitor,
-            DatabaseHealth databaseHealth,
+    public NeoStoreDataSource( File storeDir, Config config, IdGeneratorFactory idGeneratorFactory,
+            LogService logService, JobScheduler scheduler, TokenNameLookup tokenNameLookup,
+            DependencyResolver dependencyResolver, PropertyKeyTokenHolder propertyKeyTokens,
+            LabelTokenHolder labelTokens, RelationshipTypeTokenHolder relationshipTypeTokens,
+            StatementLocksFactory statementLocksFactory, SchemaWriteGuard schemaWriteGuard,
+            TransactionEventHandlers transactionEventHandlers, IndexingService.Monitor indexingServiceMonitor,
+            FileSystemAbstraction fs, TransactionMonitor transactionMonitor, DatabaseHealth databaseHealth,
             PhysicalLogFile.Monitor physicalLogMonitor,
             TransactionHeaderInformationFactory transactionHeaderInformationFactory,
-            StartupStatisticsProvider startupStatistics,
-            Guard guard,
-            CommitProcessFactory commitProcessFactory,
-            AutoIndexing autoIndexing,
-            PageCache pageCache,
-            ConstraintSemantics constraintSemantics,
-            Monitors monitors,
-            Tracers tracers,
-            Procedures procedures,
-            IOLimiter ioLimiter,
-            AvailabilityGuard availabilityGuard,
-            SystemNanoClock clock,
-            AccessCapability accessCapability,
-            StoreCopyCheckPointMutex storeCopyCheckPointMutex,
-            IdController idController )
+            StartupStatisticsProvider startupStatistics, Guard guard, CommitProcessFactory commitProcessFactory,
+            AutoIndexing autoIndexing, PageCache pageCache, ConstraintSemantics constraintSemantics, Monitors monitors,
+            Tracers tracers, Procedures procedures, IOLimiter ioLimiter, AvailabilityGuard availabilityGuard,
+            SystemNanoClock clock, AccessCapability accessCapability, StoreCopyCheckPointMutex storeCopyCheckPointMutex,
+            IdController idController, VersionContextSupplier versionContextSupplier )
```

Comment:
It becomes harder to see what's added or removed, when long parameter lists are collapsed like this.

Reply:
well, the problem with that formatting is when you do not really introduce new parameter manually it's reformatted, but will return it to its previous form here since 3.2 and things...